test_scripts/API/Additional_Submenus/004_delete_top_level_submenu_with_nested_submenus_and_commands.lua

```diff
@@ -0,0 +1,128 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal:
+-- https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0148-template-additional-submenus.md#backwards-compatibility
+-- Description: Tests sending a parentID param in a AddSubMenu request
+-- In case:
+-- 1) Mobile application is set to appropriate HMI level and System Context MENU, MAIN
+-- 2) Mobile application sends 3 AddSubMenu requests where each submenu is nested in the previous submenu
+-- 3) Mobile sends 2 Add Command requests to populate the nested submenus
+-- 4) Mobile requests to delete the top level submenu
+-- SDL does:
+-- 1) Sends DeleteSubmenu and DeleteCommand requests for the requested submenu and all contents under that submenu
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/Additional_Submenus/additional_submenus_common')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local mobileAddSubMenuRequestParams = {
+    {
+        menuID = 99, 
+        menuName = "SubMenu2",
+        parentID = 1
+    },
+    {
+        menuID = 101, 
+        menuName = "SubMenu3",
+        parentID = 99
+    }
+}
+
+local mobileAddCommandRequestParans = {
+    {
+        cmdID = 44,
+        menuParams = {
+            parentID = mobileAddSubMenuRequestParams[1].menuID,
```

Comment:
minor thing, but a possible way to make this cleaner would be to have a list of the menu and command IDs at the top of the file and reference those:
```
menuIDs = { 1, 99, 101 }
cmdIDs = { 44, 45 }
```